Sources/Concrete/Controller/HitsController.swift

```diff
@@ -87,6 +87,7 @@ extension HitsController: UITableViewDelegate {
         let hit = viewModel.hitForRow(at: indexPath)
         
         tableDelegate?.tableView(tableView, didSelectRowAt: indexPath, containing: hit)
+        viewModel.captureClickAnalyticsForHit(at: indexPath)
```

Comment:
I would capture the click before the delegate method here. 
Imagine a scenario where: 
onClickOfRow -> moveToNewVC and destroyCurrentVC (for some reason) -> HitsController gets deallocated -> the clickEvent doesn't get executed?